addons/account_peppol/wizard/account_move_send.py

```diff
@@ -0,0 +1,177 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from base64 import b64encode
+
+from odoo import api, fields, models, _
+from odoo.addons.account_edi_proxy_client.models.account_edi_proxy_user import AccountEdiProxyError
+from odoo.exceptions import UserError
+
+
+class AccountMoveSend(models.Model):
+    _inherit = 'account.move.send'
+
+    checkbox_send_peppol = fields.Boolean(
+        string='Send via PEPPOL',
+        help='Send the invoice via PEPPOL',
+    )
+    peppol_proxy_state = fields.Selection(related='company_id.account_peppol_proxy_state')
+    peppol_warning = fields.Char(
+        string="Warning",
+        compute="_compute_peppol_warning",
+    ) # technical field needed for computing a warning text about the peppol configuration
+
+    # -------------------------------------------------------------------------
+    # COMPUTE METHODS
+    # -------------------------------------------------------------------------
+
+    @api.depends('checkbox_send_peppol')
+    def _compute_checkbox_ubl_cii_xml(self):
+        # Extends account.move.send wizard - auto check checkbox_ubl_cii_xml
+        # if checkbox_send_peppol is selected
+        super()._compute_checkbox_ubl_cii_xml()
+        for wizard in self:
+            if wizard.checkbox_send_peppol and not wizard.checkbox_ubl_cii_xml:
+                wizard.checkbox_ubl_cii_xml = True
+
+    @api.depends('checkbox_send_peppol')
+    def _compute_mail_attachments_widget(self):
+        # EXTENDS 'account' - add depends
+        super()._compute_mail_attachments_widget()
+
+    @api.depends('move_ids')
+    def _compute_peppol_warning(self):
+        for wizard in self:
+            invalid_partners = wizard.move_ids.partner_id.filtered(
+                lambda partner: not partner.account_peppol_is_endpoint_valid or partner.ubl_cii_format not in ('ubl_bis3', 'xrechnung', 'nlcius', 'ubl_a_nz', 'ubl_sg'))
```

Comment:
I don't get the point of excluding all of those formats...

Reply:
It is not valid if the format is not one of those: it is whitelisting the available formats for Peppol

Reply:
```py
        selection=[
            ('facturx', "Factur-X (CII)"),
            ('ubl_bis3', "Peppol BIS Billing 3.0"),
            ('xrechnung', "XRechnung CIUS"),
            ('nlcius', "NLCIUS"),
            ('ubl_a_nz', "Peppol BIS Billing 3.0 A-NZ"),
            ('ubl_sg', "Peppol BIS Billing 3.0 SG"),
        ],
```
why not a blacklist then if we only want to exclude `facturx` ?

Reply:
because if we add other formats that shouldn't be included, we need to remember about blacklisting them everywhere in account_peppol

Reply:
We are only adding format to include... So exclude facturx instead